cupy/core/core.pyx

```diff
@@ -735,6 +735,17 @@ cdef class ndarray:
         """
         return _statistics._ndarray_argmax(self, axis, out, dtype, keepdims)
 
+    cpdef ndarray nanargmax(self, axis=None, out=None, dtype=None,
```

Comment:
Could you rename to `_nanargmax` because `numpy.ndarray` does not have `nanargmax` as an attribute?

Reply:
I am not sure why is this required? As I was having the same error thrown as `numpy.ndarray` does not have `nanargmax` as an attribute but the issue resolved after rebooting, clean building the package.

Reply:
Fix it to keep the same interface with NumPy.

Reply:
Alright will do. Thanks for the suggestion

Reply:
Sorry but renaming it to `_nanargmax` will require it to be imported as `_nanargmax` as well, right?. Other functions like `argmin` and `argmax` do not have any _ prefixes in core.pyx and core.pyd?
`>>> import cupy
>>> cupy.core.core.ndarray.nanargmax
<method 'nanargmax' of 'cupy.core.core.ndarray' objects>
>>> cupy.core.core.ndarray.nanargmin
<method 'nanargmin' of 'cupy.core.core.ndarray' objects>
>>> from cupy.sorting.search import nanargmin
>>> from cupy.sorting.search import nanargmax
>>> 
` 
I can import it now as is like this and still works. What am I missing here?

Reply:
Please rename functions also in `core.pxd`.

Reply:
After renaming in both files and rebuilding, the method is callable like 
```
>>> cupy.core.core.ndarray._nanargmax
```
Is this how we want it?

Reply:
Conventionally, single leading underscores are used for declaring private functions. So it's expected behavior.

Reply:
Gotcha. I was following the convention as defined from other functions in the file. Thanks for the clarification. I have made all the changes now. Please review